src/push/index.nativescript.ts

```diff
@@ -0,0 +1,69 @@
+import { messaging, Message } from 'nativescript-plugin-firebase/messaging';
+import { device } from 'tns-core-modules/platform';
+import { formatKinveyUrl } from '../http/utils';
+import { KinveyRequest, RequestMethod } from '../http/request';
+import { Auth } from '../http/auth';
+import { get as getConfig } from '../kinvey/config';
+
+const PUSH_NAMESPACE = 'push';
+
+export async function register(callback: (message: Message) => void, options: any = {}) {
```

Comment:
After creating the PR I realized I should change the callback parameter to be more generic. By making it more generic it makes the code sharing story easier. What do you think?

Reply:
It is a little different for the existing code. Right now you call `Kinvey.Push.register()` to register the device with Kinvey. You would not pass your callback to the register call. Instead you would use `Kinvey.Push.onNotification()` to register your callback. The above is changing that to do it in one step. However, the callback signature is the same in that it will receive one parameter that contains the data for the push notification.

Reply:
See https://github.com/Kinvey/js-sdk/blob/master/packages/kinvey-nativescript-sdk/src/push/common.ts#L26.

Reply:
I went ahead and made the callback parameter type more generic for now.